### PR TITLE
Dicom seg

### LINKED ISSUE #6: [Feat] segmentation label needs to support the DCMQI SEG-SCHEMA
Before nifti to dcm conversion ->
- [x] Have a precheck for the tilt json
- [x] have a precheck for DCMQI Schema

### PR DESCRIPTION
**Pull request by a-parida12**
Solves #6 

- json schema verification for the  dcmqii validator.

### CODE REVIEW

File: nekton/dcm2nii.py [hunk 1]
```diff
@@ -61,27 +66,89 @@ def check_slice_thickness_variable(all_dcm_paths: List[str]) -> bool:
         )
         return False if len(all_slice_thickness) == 1 else True
 
-    def _run_conv_variable(self, all_dcm_paths: List[str], name: str) -> str:
+    def _run_conv_variable(self, dicom_directory: str) -> List[str]:
```
Comment (yigitsoy) [inline nekton/dcm2nii.py]: You can use `Path` instead of str, also for the return type, there are other various places where you can do the same.

Comment (a-parida12) [reply nekton/dcm2nii.py]: will do that for all the places where string paths are expected.

File: nekton/dcm2nii.py [hunk 2]
```diff
@@ -61,27 +69,91 @@ def check_slice_thickness_variable(all_dcm_paths: List[str]) -> bool:
...
+            os.path.basename(file_path).split("_")[1:] for file_path in inp_file_list
+        ]
 
-    def run(self, dicom_directory: str, name: str) -> str:
+        out_file_list = []
+
+        # rename files
+        for i, file_path in enumerate(inp_file_list):
+            fname = (
+                name + "_" + "_".join(dcm2niix_suffix[i])
```
Comment (yigitsoy) [inline nekton/dcm2nii.py:103]: This naming convention is not clear, maybe give an example as comment.

File: nekton/dcm2nii.py [hunk 3]
```diff
@@ -61,27 +69,91 @@ def check_slice_thickness_variable(all_dcm_paths: List[str]) -> bool:
         )
```
Comment (yigitsoy) [inline nekton/dcm2nii.py:69]: Maybe you can use read_file and stop before pixel, it is a pydicom feature I guess, to just read the headers.

File: nekton/nii2dcm.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        supported_output_formats = ["dicomseg"]
+        if out_dcm_format in supported_output_formats:
+            self.format = out_dcm_format
+        else:
+            raise NotImplementedError(
+                f"""for varaible `out_dcm_format` {out_dcm_format} format not implemented.
+                 Choose one from {supported_output_formats}"""
+            )
+        supported_seg_types = ["multiclass", "multilabel"]
```
Comment (yigitsoy) [inline nekton/nii2dcm.py:24]: you can define an Enum before the class and use it also the argument in the input, similarly for the output dicom format.

File: nekton/utils/fileops.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    Raises:
+        RuntimeError: Conversion failed
+
+    Returns:
+        str: full path to the new file
+    """
+    try:
+        new_name_wo_ext = os.path.splitext(new_name)[0]
+        file_name_wo_ext = os.path.splitext(os.path.basename(complete_path))[0]
```
Comment (yigitsoy) [inline nekton/utils/fileops.py:19]: When you use Path, splitting name, base or extension is easier

File: nekton/utils/json_helpers.py [hunk 6]
```diff
@@ -37,3 +68,19 @@ def read_json(path: str) -> dict:
...
+    try:
+        with open(path, "r") as openfile:
+            data = json.load(openfile)
+    except Exception:
+        raise NameError(f"Invalid Json at {path}")
+
+    validator = _create_validator()
+
+    if not validator.is_valid(data):
+        raise NotImplementedError(
```
Comment (yigitsoy) [inline nekton/utils/json_helpers.py]: Well this is not really not implemented error, it is validation error.

File: tests/test_0_utils.py [hunk 7]
```diff
@@ -52,3 +58,72 @@ def test_0_3_check_dicom(site_package_path):
...
+    path_dcms = os.path.join(
+        site_package_path, "pydicom/data/test_files/dicomdirtests/98892001/CT5N/"
+    )
+    make_exec_bin()
+    run_bin(path_dcms)
+    out_path = os.path.join(
+        path_dcms, "CT5N_SmartScore_-_Gated_0.5_sec_20010101000000_5.nii.gz"
+    )
+    assert os.path.exists(out_path)
+    os.remove(out_path)
```
Comment (yigitsoy) [inline tests/test_0_utils.py:95]: also cleanup before run_bin, in case it remained from prev run, for some reason.

**Review by yigitsoy — CHANGES_REQUESTED** [review]
It looks pretty good. There are a few places for improvement though.

File: nekton/nii2dcm.py (hunk 4 above)
Comment (a-parida12) [reply nekton/nii2dcm.py:24]: Enum is a good idea! This format though will not be used further.

Comment (a-parida12) [reply nekton/nii2dcm.py:24]: completely getting rid of this. The idea is to have a`Nii2DcmSeg` class with 2 different methods for `multilabel`(de-prioritised for a later date) and `multiclass`.

**Pull request merged**